### PR TITLE
fix: rewritten loader logic so it orders owners by pets

### PR DESCRIPTION
**Pull request by GergoHorvathNearForm**
Closes https://github.com/nearform/the-graphql-workshop/issues/485.

The changes make the queried owners paired with the pets.

### CODE REVIEW

**Review by simoneb — COMMENTED** [review]
can we also have a test for this? one which ideally breaks with the previous implementation and succeeds with the new one

**Review by simoneb — COMMENTED** [review]
Looks better, thanks. I assume there isn't a way to preserve the ordering in the sql query, instead of having to do it in code?

File: slides.md [hunk 1]
```diff
@@ -468,7 +468,10 @@ const loaders = {
   Pet: {
     async owner(queries, context) {
       const petNames = queries.map(({ obj }) => obj.name)
-      return ownersByPetNames(context.app.pg, petNames)
+      const owners = await ownersByPetNames(context.app.pg, petNames)
+      return queries.map(({ obj: pet }) =>
+        owners.find(owner => owner.id === pet.owner)
+      )
```
Comment (simoneb) [inline slides.md:474]: if we order in sql, do we still need to order in code?

Comment (GergoHorvathNearForm) [reply slides.md]: Sorry, noticed just after pushing. Fixed.

**Review by simoneb — APPROVED** [review]

**Pull request merged**